### PR TITLE
Allow settings UI with large tab__content to appropriately allow scrolling

### LINKED ISSUE #923: Large table inside tab content doesn't have horizontal scroll bar
**Describe the bug**
When the window is small, and there is a table inside tab__content, there is no horizontal scroll bar present to see the off-screen information.

**To Reproduce**
1. Make the browser small.
2. Attempt to scroll the table.

**Expected behavior**
You should be able to scroll across the page to see the addtional information in the table.

**Screenshots**
screen shot 2018-11-12 at 17 48 32

**Desktop (please complete the following information):**
 - OS: OSX
 - Browser: Chrome
 - Version: Latest (70)

**Additional context**
This is likely caused by the tab__content being rendered using `display: table`. And then placing a table inside this.

### LINKED ISSUE #932: Improve radio buttons in action templates
Clicking on a radio button the orange helper appears but looks wrong.

1. Go to 'Rules' page
2. Create a new  'Export as file' rule
3. Fill out the tabs until 'Fields to Export'
4. On the 'Fields to include' select one.

**Expected behavior**
Better styling of the orange square on radio button

screenshot 2019-01-04 at 10 52 40

**Desktop (please complete the following information):**
 - OS: mac OS 10.14.2
 - Browser Chrome
 - Version 71.0.3578.98

**Additional context**
Could happen in other browsers.

### PR DESCRIPTION
**Pull request by Stanton**
See #923 for gifs showing cross browser tests.

This change updates the settings layout pattern to use flexbox for the right hand column, which allows horizontally scrolling tables to correctly apply their expected scrolling/overflow behaviour.

This issue was observed in Jadu XFP FormBuilder, and the same pattern is used in Jadu CMS Galaxies Designer.

flex-chrome

For the issue reported, IE10, IE11, Edge, Chrome, Safari, Firefox now all allow the table to scroll (providing the table contains the `.table--horizontal` class).

IE9 scrolls the entire `tab__inner`.

* Tested with an approximation of the affected XFP UI within Pulsar
* Tested with the Galaxies designer prototype

Closes #923 

## Breaking change checklist

The following points should be considered as an early-warning of introducing a breaking change to a Continuum platform product. This is not an exhaustive list of what constitutes a breaking change.

| Does this pull request... | Yes / No |
| ---- | ---- |
| require changes to `main.js` | ✅ No |
| require changes to `index.js` | ✅ No |
| require changes to `pulsar.scss` | ✅ No |
| require changes to `gruntfile.js` | ✅ No |
| require changes to `package.json` (not including dev dependencies) | ✅ No |
| require changes to `base.html.twig` (or other core views like header/footer) | ✅ No |
| require new arguments to be passed to a js component | ✅ No |
| require markup changes to maintain functionality | ✅ No |
| require changes to existing unit tests | ✅ No |

## Browser testing checklist

- [x] Chrome
- [x] Safari
- [x] Firefox
- [x] Edge
- [x] IE 11
- [x] IE 10
- [x] IE 9
- [x] Mobile Safari
- [x] Samsung Internet

### CODE REVIEW

File: stylesheets/_layout.settings.scss [hunk 1]
```diff
@@ -352,16 +358,31 @@
...
+
 .tab__container.has-settings .tab__content {
     @include respond-min($screen-medium) {
         margin-bottom: $line-height-small;
-        padding-left: $padding-small-horizontal * 2;
+        -webkit-flex: 1 1 auto;
+        -moz-flex: 1 1 auto;
+        -ms-flex: 1 1 auto;
+        -o-flex: 1 1 auto;
+        flex: 1 1 auto;
```
Comment (jackbentley) [inline stylesheets/_layout.settings.scss]: ❓ What are these for?

**Comment by jackbentley** [comment]
❓ Noticed in the gif you posted there was an odd problem with the background of the settings sidebar on the second to last breakpoint?

File: stylesheets/_layout.settings.scss [hunk 2]
```diff
@@ -371,7 +392,13 @@
 
 .tab__container.has-settings .tab__inner {
     @include respond-min($screen-desktop) {
+        align-items: flex-start;
         display: table;
+        display: -webkit-box;
+        display: -moz-box;
+        display: -ms-flexbox;
+        display: -webkit-flex;
+        display: flex;
```
Comment (jackbentley) [inline stylesheets/_layout.settings.scss]: 👍 Yay! Flex!

File: stylesheets/_layout.settings.scss (hunk 1 above)
Comment (Stanton) [reply stylesheets/_layout.settings.scss]: The prefixes or the longhand?

Comment (jackbentley) [reply stylesheets/_layout.settings.scss]: The usage of `flex` 😄

Comment (mikedevelops) [reply stylesheets/_layout.settings.scss]: The autoprefixer should take care of the vendor stuff shouldn't it?

Comment (Stanton) [reply stylesheets/_layout.settings.scss]: Found this, it's a weird bug in IE10 which caused the table to extend even further at wide viewport widths. Moving flex to the container doesn't resolve the issue, but I've removed all but `-ms-flex` to just target IE with this.

**Before**
ie10-before

**After**
ie10-after

File: stylesheets/_layout.settings.scss (hunk 2 above)
Comment (mikedevelops) [inline stylesheets/_layout.settings.scss]: More vendor here that we should be able to remove ☝️

File: stylesheets/_layout.settings.scss (hunk 1 above)
Comment (jackbentley) [reply stylesheets/_layout.settings.scss]: Very weird

File: stylesheets/_layout.settings.scss (hunk 2 above)
Comment (jackbentley) [reply stylesheets/_layout.settings.scss]: Providing the auto-prefixer takes care of this?

Comment (Stanton) [reply stylesheets/_layout.settings.scss]: It does, they've been removed

File: stylesheets/_layout.settings.scss [hunk 3]
```diff
@@ -371,7 +391,9 @@
 
 .tab__container.has-settings .tab__inner {
     @include respond-min($screen-desktop) {
+        align-items: flex-start;
         display: table;
```
[the remark this answers is not in the document]
Comment (Stanton) [reply stylesheets/_layout.settings.scss]: I'm not going to change that as part of this ticket for the following reasons:

* dIsplay:table was pre-existing and not part of this PR
* It took ages to test the current set of changes

**Pull request merged**